aeron-cluster/src/main/java/io/aeron/cluster/ConsensusModuleAgent.java

```diff
@@ -1373,6 +1373,24 @@ int addLogPublication()
                 recoveryPlan.appendedLogPosition, recoveryPlan.log.initialTermId, recoveryPlan.log.termBufferLength);
             channelUri.put(MTU_LENGTH_PARAM_NAME, Integer.toString(recoveryPlan.log.mtuLength));
         }
+        else
+        {
+            if (!channelUri.containsKey(INITIAL_TERM_ID_PARAM_NAME))
+            {
+                channelUri.put(INITIAL_TERM_ID_PARAM_NAME, "0");
+            }
+
+            if (!channelUri.containsKey(TERM_ID_PARAM_NAME))
+            {
+                channelUri.put(TERM_ID_PARAM_NAME, "0");
+            }
+
+            if (!channelUri.containsKey(TERM_OFFSET_PARAM_NAME))
+            {
+                channelUri.put(TERM_OFFSET_PARAM_NAME, "0");
+            }
+        }
+
```

Comment:
Shouldn't these be set together for a correct result? For example if the `init-term-id` is set to `87` but the `term-id` is not set this code will set `term-id=0` which will break.

Could we check if any of the values is missing and set all three to zero at that point?

Reply:
These values should not be set in the template. We should error if they are.